src/main/java/org/generationcp/middleware/dao/dms/ObservationUnitsSearchDao.java

```diff
@@ -61,7 +61,9 @@ public class ObservationUnitsSearchDao extends GenericDAO<ExperimentModel, Integ
 	private static final String GERMPLASM_JOIN = " INNER JOIN germplsm g on g.gid = s.dbxref_id ";
 	private static final String IMMEDIATE_SOURCE_NAME_JOIN = " LEFT JOIN names immediateSource  ON g.gpid2 = immediateSource.gid AND immediateSource.nstat = 1 ";
 	private static final String GROUP_SOURCE_NAME_JOIN =
-		"LEFT JOIN names groupSourceName ON groupSourceName.gid = g.gpid1 AND g.gnpgs < 0";
+		" LEFT JOIN names groupSourceName ON groupSourceName.gid = g.gpid1 AND g.gnpgs < 0";
```

Comment:
@mderamos I am ok reusing this query, at the end both are datasets. I did not run it locally but I am interested to know if we are just returning the needed columns and no extra information that it is associated to the plot dataset (observations table), and that of course it is not shown in the frontend because we just show the info for the columns contained in /table service
Could you please check or share the query that is run when the service is called from a MEANS dataset

Reply:
Hi @clarysabel , here's the exact query. No extra information that it is associated to the plot dataset is returned on the query. Only data that is relevant to the means observations have value. 

SELECT * FROM (SELECT      (SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8582) AS 'COL',     s.dbxref_id AS 'GID',     (SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8410) AS 'FIELDMAP RANGE',     coalesce(nullif((SELECT distinct(lot.stock_id)         FROM ims_experiment_transaction  ndt 			inner join ims_transaction tr on tr.trnid = ndt.trnid and ndt.type = 1 and tr.trntype = 1			inner join ims_lot lot on lot.lotid = tr.lotid 
        WHERE ndt.nd_experiment_id = nde.nd_experiment_id), '-') ) AS 'STOCK_ID',     coalesce(nullif((SELECT count(sp.sample_id)         FROM sample sp         WHERE sp.nd_experiment_id = nde.nd_experiment_id)          + coalesce(child_sample_count.count, 0), 0), '-') AS 'SUM_OF_SAMPLES',     nde.nd_experiment_id as observationUnitId,     (SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8400) AS 'FIELDMAP COLUMN',     parent.obs_unit_id as PARENT_OBS_UNIT_ID,     nde.obs_unit_id AS 'OBS_UNIT_ID', (select group_concat(fcvt.cvterm_id separator ',') from file_metadata fm  inner join file_metadata_cvterm fcvt on fm.file_id = fcvt.file_metadata_id where fm.nd_experiment_id = nde.nd_experiment_id) as 'FILE_TERM_IDS',     (SELECT edesign.name FROM nd_geolocationprop gprop INNER JOIN cvterm edesign on edesign.cvterm_id = gprop.value WHERE gprop.nd_geolocation_id = gl.nd_geolocation_id and gprop.type_id = 8135) AS 'EXPT_DESIGN',     s.cross_value AS 'CROSS',     s.name AS 'DESIGNATION',     gl.description AS 'TRIAL_INSTANCE',     (SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8220) AS 'BLOCK_NO',     (SELECT loc.lname FROM nd_geolocationprop gprop INNER JOIN location loc on loc.locid = gprop.value WHERE gprop.nd_geolocation_id = gl.nd_geolocation_id and gprop.type_id = 8190) AS 'LOCATION_ID',     (SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8581) AS 'ROW',     (SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8200) AS 'PLOT_NO',     (SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8210) AS 'REP_NO', (select count(1) from file_metadata fm where fm.nd_experiment_id = nde.nd_experiment_id) as 'FILE_COUNT',  MAX(IF(cvterm_variable.name = 'GW_DW_g1000grn_Means', ph.value, NULL)) AS 'GW_DW_g1000grn_Means', MAX(IF(cvterm_variable.name = 'GW_DW_g1000grn_Means', ph.phenotype_id, NULL)) AS 'GW_DW_g1000grn_Means_PhenotypeId', MAX(IF(cvterm_variable.name = 'GW_DW_g1000grn_Means', ph.status, NULL)) AS 'GW_DW_g1000grn_Means_Status', MAX(IF(cvterm_variable.name = 'GW_DW_g1000grn_Means', ph.cvalue_id, NULL)) AS 'GW_DW_g1000grn_Means_CvalueId',  MAX(IF(cvterm_variable.name = 'GW_DW_g1000grn_Means', ph.draft_value, NULL)) AS 'GW_DW_g1000grn_Means_DraftValue', MAX(IF(cvterm_variable.name = 'GW_DW_g1000grn_Means', ph.draft_cvalue_id, NULL)) AS 'GW_DW_g1000grn_Means_DraftCvalueId',  MAX(IF(cvterm_variable.name = 'GW_DW_g100grn_Means', ph.value, NULL)) AS 'GW_DW_g100grn_Means', MAX(IF(cvterm_variable.name = 'GW_DW_g100grn_Means', ph.phenotype_id, NULL)) AS 'GW_DW_g100grn_Means_PhenotypeId', MAX(IF(cvterm_variable.name = 'GW_DW_g100grn_Means', ph.status, NULL)) AS 'GW_DW_g100grn_Means_Status', MAX(IF(cvterm_variable.name = 'GW_DW_g100grn_Means', ph.cvalue_id, NULL)) AS 'GW_DW_g100grn_Means_CvalueId',  MAX(IF(cvterm_variable.name = 'GW_DW_g100grn_Means', ph.draft_value, NULL)) AS 'GW_DW_g100grn_Means_DraftValue', MAX(IF(cvterm_variable.name = 'GW_DW_g100grn_Means', ph.draft_cvalue_id, NULL)) AS 'GW_DW_g100grn_Means_DraftCvalueId',  MAX(IF(cvterm_variable.name = 'GW_DW_g200grn_Means', ph.value, NULL)) AS 'GW_DW_g200grn_Means', MAX(IF(cvterm_variable.name = 'GW_DW_g200grn_Means', ph.phenotype_id, NULL)) AS 'GW_DW_g200grn_Means_PhenotypeId', MAX(IF(cvterm_variable.name = 'GW_DW_g200grn_Means', ph.status, NULL)) AS 'GW_DW_g200grn_Means_Status', MAX(IF(cvterm_variable.name = 'GW_DW_g200grn_Means', ph.cvalue_id, NULL)) AS 'GW_DW_g200grn_Means_CvalueId',  MAX(IF(cvterm_variable.name = 'GW_DW_g200grn_Means', ph.draft_value, NULL)) AS 'GW_DW_g200grn_Means_DraftValue', MAX(IF(cvterm_variable.name = 'GW_DW_g200grn_Means', ph.draft_cvalue_id, NULL)) AS 'GW_DW_g200grn_Means_DraftCvalueId',  COALESCE(nde.observation_unit_no, (		SELECT ndep.value FROM nd_experimentprop ndep INNER JOIN cvterm ispcvt ON ispcvt.cvterm_id = ndep.type_id WHERE ndep.nd_experiment_id = plot.nd_experiment_id AND ndep.type_id = 8200  )) AS OBSERVATION_UNIT_NO FROM 	project p 	INNER JOIN nd_experiment nde ON nde.project_id = p.project_id 	INNER JOIN nd_geolocation gl ON nde.nd_geolocation_id = gl.nd_geolocation_id 	LEFT JOIN stock s ON s.stock_id = nde.stock_id  LEFT JOIN stockprop sp ON sp.stock_id = s.stock_id  LEFT JOIN cvterm cvterm_entry_variable ON (cvterm_entry_variable.cvterm_id = sp.type_id) 	LEFT JOIN phenotype ph ON nde.nd_experiment_id = ph.nd_experiment_id 	LEFT JOIN cvterm cvterm_variable ON cvterm_variable.cvterm_id = ph.observable_id  LEFT JOIN nd_experiment parent ON parent.nd_experiment_id = nde.parent_id  LEFT JOIN (SELECT parent.nd_experiment_id,        nullif(count(child_sample.sample_id), 0) AS count      FROM nd_experiment child             LEFT JOIN sample child_sample ON child.nd_experiment_id = child_sample.nd_experiment_id             INNER JOIN nd_experiment parent ON child.parent_id = parent.nd_experiment_id      GROUP BY parent.nd_experiment_id) child_sample_count ON child_sample_count.nd_experiment_id = nde.nd_experiment_id  INNER JOIN nd_experiment plot ON plot.nd_experiment_id = nde.parent_id OR ( plot.nd_experiment_id = nde.nd_experiment_id and nde.parent_id is null )  WHERE p.project_id = 25015 GROUP BY nde.nd_experiment_id  ) T ORDER BY (1 * `TRIAL_INSTANCE`) ASC, `TRIAL_INSTANCE` ASC

Reply:
Thanks for sharing, I can see that the query still returns some information related to the observations tab, as it was designed only for the obs and sub-obs tab.
I really like the idea of consolidating only one query for all dataset types but we should find a way to avoid returning info that is not required or it is not part of the given dataset 
For example, in the query that you have shared, we can see COL, FIELDMAP RANGE, STOCK_ID, SUM_OF_SAMPLES, FIELDMAP COLUMN, OBS_UNIT_ID and many others that are not really part of a means dataset
I think this is not the right time to change such a core query, we are very close to the release. Could you please create a technical ticket to improve it during 22.1 sprint and share it with me so I make sure we give it the right priority? 
I will merge this PR and wait for the new ticket! 
Thanks 
CC @nahuel-soldevilla